### DISCUSSION TITLE
运行tests的时候，有办法看到每个peer的输出吗？

### DISCUSSION THREAD

**Discussion by OsmiumD** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#51] [Q&A]
在运行tests的时候，pycharm里面只会输出test的结果和test过程中的print，而不会输出里面每个peer的print。请问有办法可以看到运行tests时候每个或者某个peer的console output来方便debug吗？

**Comment by Godblessmycode1** [comment]
不运行pytest,用terminal进行模拟，可以看到输出

**Reply by OsmiumD** [reply]
但是在transfer_test里面，如果不用pytest的那个模拟，是不会出现掉包的

**Reply by TANSixu** [reply]
hi 感谢回答！
是的 可以通过terminal运行来看到stdout。此外，可以使用simsocket中的add log方法将信息打印到log里。如果实在需要test的stdout信息，可以修改grader中启动peer process的命令，将子进程的stdout指向文件。如需教程或指导请再联系我。

**Reply by LoserXj** [reply]
想问下这里的将子进程指向文件是怎么具体指向呢

**Reply by TANSixu** [reply]
> 想问下这里的将子进程指向文件是怎么具体指向呢

hi，已发布到discussion，可参考 #58
